backend/dataall/modules/omics/services/omics_service.py

```diff
@@ -0,0 +1,179 @@
+    description: str = 'No description provided'
+    tags: List[str] = field(default_factory=list)
+    S3InputBucket: str = 'No input bucket provided'
+    S3InputPrefix: str = ''
+    S3OutputBucket: str = 'No output bucket provided'
+    S3OutputPrefix: str = ''
+
+    @classmethod
+    def from_dict(cls, env):
+        """Copies only required fields from the dictionary and creates an instance of class"""
+        fields = set([f.name for f in dataclasses.fields(cls)])
+        return cls(**{k: v for k, v in env.items() if k in fields})
+
+
+class OmicsService:
+    """
+    Encapsulate the logic of interactions with Omics.
+    """
+
+    @staticmethod
+    @TenantPolicyService.has_tenant_permission(MANAGE_OMICS_RUNS)
+    @ResourcePolicyService.has_resource_permission(CREATE_OMICS_RUN)
+    @GroupPolicyService.has_group_permission(CREATE_OMICS_RUN)
+    def create_omics_run(*, uri: str, admin_group: str, data: dict) -> OmicsRun:
+        """
+        Creates an omics_run and attach policies to it
+        Throws an exception if omics_run are not enabled for the environment
+        """
+
+        with _session() as session:
+            environment = EnvironmentService.get_environment_by_uri(session, uri)
+            dataset = DatasetRepository.get_dataset_by_uri(session, data['destination'])
+            enabled = EnvironmentService.get_boolean_env_param(session, environment, 'omicsEnabled')
+
+            if not enabled and enabled.lower() != 'true':
+                raise exceptions.UnauthorizedOperation(
+                    action=CREATE_OMICS_RUN,
+                    message=f'OMICS_RUN feature is disabled for the environment {environment.label}',
+                )
+
+            omics_run = OmicsRun(
+                owner=get_context().username,
+                organizationUri=environment.organizationUri,
+                environmentUri=environment.environmentUri,
+                SamlAdminGroupName=admin_group,
+                workflowUri=data['workflowUri'],
+                parameterTemplate=data['parameterTemplate'],
+                label=data['label'],
+                outputUri=f's3://{dataset.S3BucketName}',
+                outputDatasetUri=dataset.datasetUri,
+            )
+
+            OmicsRepository(session).save_omics_run(omics_run)
+
+            response = OmicsClient.run_omics_workflow(omics_run, session)
+            omics_run.runUri = response['id']
+            ResourcePolicyService.attach_resource_policy(
+                session=session,
+                group=omics_run.SamlAdminGroupName,
+                permissions=OMICS_RUN_ALL,
+                resource_uri=omics_run.runUri,
+                resource_type=OmicsRun.__name__,
+            )
+            OmicsRepository(session).save_omics_run(omics_run)
+
+            return omics_run
+
+    @staticmethod
+    @ResourcePolicyService.has_resource_permission(GET_OMICS_RUN)
+    def get_omics_run(*, uri: str):
```

Comment:
Unused method, if not needed then we can delete. If we delete it then we also need to check if the permission GET_OMICS_RUN is used anywhere. If not, we delete it